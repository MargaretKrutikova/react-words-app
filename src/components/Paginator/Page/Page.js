import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Page extends PureComponent {
  setPage = (event, page) => {
    if (event) {
      event.preventDefault();
    }
    if (this.props.isDisabled || this.props.isActive) return;
    this.props.onPageChange(page);
  }
  getPageClass() {
    let classes = ['page-item'];
    this.props.isActive && classes.push('active');
    this.props.isDisabled && classes.push('disabled');

    return classes.join(' ');
  }
  render() {
    const { page } = this.props;

    return (
      <li className={this.getPageClass()}>
        <a href="#" className='page-link' onClick={(event) => this.setPage(event, page)}>
          { this.props.children || page }
        </a>
      </li>);
  }
}

Page.propTypes = {
  isActive: PropTypes.bool,
  isDisabled: PropTypes.bool,
  page: PropTypes.number.isRequired,
  children: PropTypes.node,
  onPageChange: PropTypes.func.isRequired
};

export default Page;