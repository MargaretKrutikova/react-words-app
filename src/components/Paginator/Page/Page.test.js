import 'babel-polyfill'; // required for Object.values

import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
import Page from './Page';

test('Page changes classes if isActive or isDisabled props change', () => {
  const testProps = {
    isActive: false,
    isDisabled: false,
    page: 4,
    onPageChange: () => {}
  };
  const component = Enzyme.shallow( <Page {...testProps} />);
  
  expect(component).toMatchSnapshot();

  // change props and rerender
  component.setProps({ isActive: true });
  expect(component).toMatchSnapshot();

  // change props and rerender
  component.setProps({ isActive: false });
  component.setProps({ isDisabled: true });
  expect(component).toMatchSnapshot();
});