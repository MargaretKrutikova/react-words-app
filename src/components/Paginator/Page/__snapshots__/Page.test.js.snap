// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page changes classes if isActive or isDisabled props change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    isActive={false}
    isDisabled={false}
    onPageChange={[Function]}
    page={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href="#"
        onClick={[Function]}
      >
        4
      </a>,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 4,
        "href": "#",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": 4,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="#"
          onClick={[Function]}
        >
          4
        </a>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "href": "#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 4,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Page changes classes if isActive or isDisabled props change 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    isActive={true}
    isDisabled={false}
    onPageChange={[Function]}
    page={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href="#"
        onClick={[Function]}
      >
        4
      </a>,
      "className": "active",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 4,
        "href": "#",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": 4,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="#"
          onClick={[Function]}
        >
          4
        </a>,
        "className": "active",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "href": "#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 4,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Page changes classes if isActive or isDisabled props change 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    isActive={false}
    isDisabled={true}
    onPageChange={[Function]}
    page={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href="#"
        onClick={[Function]}
      >
        4
      </a>,
      "className": "disabled",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 4,
        "href": "#",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": 4,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="#"
          onClick={[Function]}
        >
          4
        </a>,
        "className": "disabled",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "href": "#",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 4,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
